tests/callbacks/test_TimerListener.cpp

```diff
@@ -0,0 +1,288 @@
+
+    ASSERT_GT(par_for_timer_matching->timer.duration(), Kokkos::utils::timer::seconds(0));
+
+    EXPECT_FALSE(par_for_timer_no_match->connected() || par_for_timer_no_match->closed())
+        << "The listener should not be connected or closed because the events did not match.";
+
+    Manager::unregister_listener(par_for_timer_matching.get());
+    Manager::unregister_listener(par_for_timer_no_match.get());
+}
+
+//! Listener to time regions whose name matches a regex.
+using region_timer_t = RegionTimerListener<EventRegexMatcher>;
+
+//! @test Check traits of @ref RegionTimerListener.
+TEST(RegionTimerListener, traits)
+{
+    /// Check that @ref region_timer_t is a listener for @ref PushRegionEvent
+    /// and @ref PopRegionEvent.
+    static_assert(std::same_as<
+        listener_event_type_list_t<region_timer_t>,
+        Kokkos::Impl::type_list<PushRegionEvent, PopRegionEvent>
+    >);
+
+    static_assert(Listener<region_timer_t>);
+}
+
+/**
+ * @test Check the behavior of @ref RegionTimerListener.
+ *
+ * @note The assertion that the duration provided by the region timer is larger
+ *       than the duration provided by the parallel for timer relies on the fact
+ *       that in the test workload, the parallel for on the non-default execution
+ *       space instance is followed by a fence and the parallel reduce on the
+ *       default space instance reduces into a host space scalar, thus ensuring
+ *       that this parallel reduce is blocking. In other words, for these reasons,
+ *       we can use a region timer without execution space and without tool fences.
+ */
+TEST_F(TimerListenerTest, region)
+{
+    const auto par_for_timer = std::make_shared<parallel_for_timer_t>(
+        std::regex("computation - level 0 - pfor"), exec
+    );
+
+    const auto region_timer = std::make_shared<region_timer_t>(
+        std::regex("computation - level 0")
+    );
+
+    Manager::register_listener(par_for_timer);
+    Manager::register_listener(region_timer);
+
+    MyWorkload<execution_space>{}.execute(exec);
+
+    ASSERT_GT(par_for_timer->timer.duration(), Kokkos::utils::timer::milliseconds{0.});
+    ASSERT_LT(par_for_timer->timer.duration(), region_timer->timer.duration());
+
+    Manager::unregister_listener(par_for_timer.get());
+    Manager::unregister_listener(region_timer.get());
+}
+
+//! @test Check traits of @ref RegionTimerListener when used with @ref EnqueuedEventTimer.
+TEST(RegionTimerListener, traits_when_used_with_enqueued_event_timer)
+{
+    using region_timer_with_enqueued_event_timer_t = RegionTimerListener<
+        EventRegexMatcher,
+        EnqueuedEventTimer<execution_space>
+    >;
+
+    static_assert(std::constructible_from<
+        region_timer_with_enqueued_event_timer_t,
+        std::regex, execution_space
+    >);
+}
+
+//! Timer that measures the time to launch, in addition to the time to execute the kernel.
+template <Kokkos::utils::concepts::ExecutionSpace Exec>
+struct EnqueuedEventTimerWithLatency
```

Comment:
Let's move this one to its own include file :wink: (To make it visible)

Reply:
OK. Done. It's PR:
- #81